### ISSUE TITLE
Inf2 optimum-neuron inference test failures

### ISSUE THREAD

**Issue by musunita** [huggingface/optimum-neuron#382]
### INF 2 Test Summary:

* Total Test Passed: 426
* Total Test Failed: 35
* Errors : 9

35 failed, 426 passed, 248 deselected, 105 warnings, 9 errors in 5530.71s (1:32:10) =

**Test Steps :**
    1. Setup the neuron environment for your new release
         a. Since the neuron SDK is now used for both inf2 and inf1, make sure to test 
        on both scenarios. 
    2.  
        Setup optimum environment 
        a. Make sure that tests pass for the latest release and the dev branch. 
         pip install optimum-neuron
    3.  Install the latest release: 2.16
        Install the dev branch:  pip3 install git+https://github.com/huggingface/optimum-neuron#egg=optimum-neuron[tests] tokenizers==0.14.1
    4. clone optimum-neuron repo for tests
        a. git clone https://github.com/huggingface/optimum-neuron.git
        b. cd optimum-neuron/
        c. pytest -m is_inferentia_test tests
    5.  
        Run tests
        a . cd optimum-neuron/
        b. pytest -m is_inferentia_test tests  

**INF2 TESTING DETAILS:**

Config: Inf2.48xlarge tests

Release version:
```
(aws_neuron_venv_pytorch) ubuntu@ip-10-4-130-63:~$ pip list | grep "neuron"
aws-neuronx-runtime-discovery 2.9
libneuronxla 0.5.570
neuron-torch-tools 1.0.0.1459+4120acb19
neuronx-cc 2.11.0.35+4f5279863
neuronx-distributed 0.5.0
neuronx-hwm 2.11.0.2+e34678757
optimum-neuron 0.0.16.dev0
torch-neuronx 1.13.1.1.12.1
torch-xla 1.13.1+torchneuronc
transformers-neuronx 0.8.268
```

**Failed Testcases:**

test_model_from_path[hf-internal-testing/tiny-random-BloomForCausalLM]
--
test_model_from_path[hf-internal-testing/tiny-random-gpt2]
test_model_from_path[dacorvo/tiny-random-llama]
test_model_from_path[hf-internal-testing/tiny-random-OPTForCausalLM]
test_decoder_generation[hf-internal-testing/tiny-random-BloomForCausalLM-sample]
test_decoder_generation[hf-internal-testing/tiny-random-BloomForCausalLM-sample-with-temp
test_decoder_generation[hf-internal-testing/tiny-random-BloomForCausalLM-greedy]
test_decoder_generation[hf-internal-testing/tiny-random-BloomForCausalLM-greedy_no-repeat]
test_model_generation_input_dimensions[hf-internal-testing/tiny-random-BloomForCausalLM]
test_decoder_generation[hf-internal-testing/tiny-random-gpt2-sample]
test_decoder_generation[hf-internal-testing/tiny-random-gpt2-sample-with-temp]
test_decoder_generation[hf-internal-testing/tiny-random-gpt2-greedy]
test_decoder_generation[hf-internal-testing/tiny-random-gpt2-greedy_no-repeat]
test_model_generation_input_dimensions[hf-internal-testing/tiny-random-gpt2]
test_decoder_generation[dacorvo/tiny-random-llama-sample]
test_decoder_generation[dacorvo/tiny-random-llama-sample-with-temp]
test_decoder_generation[dacorvo/tiny-random-llama-greedy]
test_decoder_generation[dacorvo/tiny-random-llama-greedy_no-repeat]
test_model_generation_input_dimensions[dacorvo/tiny-random-llama]
test_decoder_generation[hf-internal-testing/tiny-random-OPTForCausalLM-sample]
test_decoder_generation[hf-internal-testing/tiny-random-OPTForCausalLM-sample]
test_decoder_generation[hf-internal-testing/tiny-random-OPTForCausalLM-sample]
test_decoder_generation[hf-internal-testing/tiny-random-OPTForCausalLM-sample-with-temp]
test_decoder_generation[hf-internal-testing/tiny-random-OPTForCausalLM-greedy]
test_decoder_generation[hf-internal-testing/tiny-random-OPTForCausalLM-greedy_no-repeat]
test_model_generation_input_dimensions[hf-internal-testing/tiny-random-OPTForCausalLM]
test_model_from_hub
test_push_to_hub[hf-internal-testing/tiny-random-BloomForCausalLM]
test_push_to_hub[hf-internal-testing/tiny-random-gpt2]
test_push_to_hub[dacorvo/tiny-random-llama]
test_push_to_hub[hf-internal-testing/tiny-random-OPTForCausalLM]
test_seq2seq_model_from_hub
test_load_no_parameters[hf-internal-testing/tiny-random-gpt2]
test_from_model_and_tokenizer[hf-internal-testing/tiny-random-gpt2]
test_load_no_parameters[dacorvo/tiny-random-llama]
test_from_model_and_tokenizer[dacorvo/tiny-random-llama]
test_from_hub

Failure logs: 
```
self = DecoderLmHeadForSamplingNoEmbedding(
  (layers): ModuleList()
)

    def get_all_kernels(self):
>       return self.program.get_kernels()
E       AttributeError: 'NoneType' object has no attribute 'get_kernels'
```
```
[INF2_on_test_results.txt](https://github.com/huggingface/optimum-neuron/files/13688080/INF2_on_test_results.txt)

    @classmethod
    def setUpClass(cls):
        if os.environ.get("HF_TOKEN_OPTIMUM_NEURON_CI", None) is not None:
            token = os.environ.get("HF_TOKEN_OPTIMUM_NEURON_CI")
            HfFolder.save_token(token)
        else:
>           raise RuntimeError("Please specify the token via the HF_TOKEN_OPTIMUM_NEURON_CI environment variable.")
E           RuntimeError: Please specify the token via the HF_TOKEN_OPTIMUM_NEURON_CI environment variable.
```

Attached complete logs for the failures: 
[INF2_on_test_results.txt](https://github.com/huggingface/optimum-neuron/files/13688087/INF2_on_test_results.txt)

**Comment by dacorvo** [comment]
From your list of packages, this looks like you installed first the AWS Neuron SDK 2.16, then `optimum-neuron`, that uses fixed versions of `neuronx-cc`, `transformers-neuronx`  and `neuronx-distributed` corresponding to AWS Neuron SDK 2.15.
I think this is what explains the first error you get (probably a version mismatch between `libneuronxla` and `transformers-neuronx`).
For the hub tests, you would need a token with access to our CI staging environment. We should discuss directly how to solve this.

**Comment by dacorvo** [comment]
Fixed in `optimum-neuron` 0.0.17.
